scripts/partition_manager.py

```diff
@@ -362,22 +367,25 @@ def _set_addresses_and_align(reqs, sub_partitions, solution, size, start, dynami
             higher_partition = solution[i + 1]
             reqs[current]['address'] = reqs[higher_partition]['address'] - reqs[current]['size']
 
-        if align_if_required(i, dynamic_partitions, False, reqs, solution):
-            _set_addresses_and_align(reqs, sub_partitions, solution, size, start, dynamic_partitions, dp)
+        if align_if_required(i, dynamic_partitions, False, reqs, dp, solution):
+            try:
+                _set_addresses_and_align(reqs, sub_partitions, solution, size, start, dynamic_partitions, dp)
+            except RecursionError:
+                raise RuntimeError(ALIGNMENT_ERROR)
```

Comment:
Could we give the user a hint ?
We must know something about which partitions cannot be aligned, as example if there are too many dynamic partition with alignment reqs.
Or provide hime with some info on what we did align, and what couldn't fit.
Like:
```
Partition A; start=0, end=FF
Partition B; start=10, end=FFFFFFFF
couldn't fit partition C - No space left on device.
```
Go and buy an external flash @hakonfam webstore.

Reply:
Added comments where we know why it failed. In the generic case its not that trivial, since we might support multiple dynamic partition alignment if we are lucky.